crates/l1tx/src/filter/types.rs

```diff
@@ -1,20 +1,85 @@
+use std::cmp::Ordering;
+
+use bitcoin::{Amount, OutPoint};
 use borsh::{BorshDeserialize, BorshSerialize};
 use strata_primitives::{
     block_credential::CredRule,
     buf::Buf32,
-    l1::{BitcoinAddress, Outpoint},
+    l1::{BitcoinAddress, BitcoinScriptBuf, OutputRef},
     params::{DepositTxParams, RollupParams},
     sorted_vec::SortedVec,
 };
+use strata_state::{
+    bridge_state::{DepositEntry, DepositState},
+    chain_state::Chainstate,
+};
 
 use crate::utils::{generate_taproot_address, get_operator_wallet_pks};
 
+// TODO: This is FIXED OPERATOR FEE for TN1
+pub const OPERATOR_FEE: Amount = Amount::from_int_btc(2);
+
 #[derive(Clone, Debug, PartialEq, Eq, BorshSerialize, BorshDeserialize)]
 pub struct EnvelopeTags {
     pub checkpoint_tag: String,
     pub da_tag: String,
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, BorshSerialize, BorshDeserialize)]
+pub struct ExpectedWithdrawalFulfillment {
+    /// withdrawal address scriptbuf
+    pub destination: BitcoinScriptBuf,
+    /// Expected minimum withdrawal amount in sats
+    pub amount: u64,
+    /// index of assigned operator
+    pub operator_idx: u32,
+    /// index of assigned deposit entry for this withdrawal
+    pub deposit_idx: u32,
+}
+
+// ordering based on destination for binary search.
+impl Ord for ExpectedWithdrawalFulfillment {
+    fn cmp(&self, other: &Self) -> Ordering {
+        self.destination.cmp(&other.destination)
+    }
+}
+
+impl PartialOrd for ExpectedWithdrawalFulfillment {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, BorshSerialize, BorshDeserialize)]
+pub struct DepositSpendConfig {
+    /// index of deposit entry
+    pub deposit_idx: u32,
+    /// utxo for this deposit
+    pub output: OutputRef,
+}
+
+impl DepositSpendConfig {
+    pub fn from_outpoint(outpoint: OutPoint) -> Self {
+        Self {
+            deposit_idx: 0,
+            output: OutputRef::from(outpoint),
+        }
+    }
+}
+
+// sort based on output field for binary search.
+impl Ord for DepositSpendConfig {
+    fn cmp(&self, other: &Self) -> Ordering {
+        self.output.cmp(&other.output)
+    }
+}
+
+impl PartialOrd for DepositSpendConfig {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
```

Comment:
Again this isn't a compliant implementation, see the prev comment.

Reply:
I've now added a new trait 
```rust 
pub trait HasKey<K: Ord> {
   fn get_key(&self) -> K 
}
```

And a new struct `SortedVecWithKey<K: Ord, T: HasKey<K>>` which doesn't require `PartialOrd` to be implemented on complex types